### PR TITLE
[EN DateTimeV2] Fixed inconsistency between English and Spanish in handling "from" and issue with list (#2287)

### PR DESCRIPTION
**Pull request by LionbridgeCSII**
Moved support for ranges like "from 2015" from Standard to Experimental mode in analogy with Spanish behaviour (#2287).

Added pattern "from 2015 and 2016"  to PotentialAmbiguousRangeRegex (FromToRegex) to avoid considering 'from' a modifier in this case.

Updated affected test cases in DateTimeModel and moved the originals to DateTimeModelExperimentalMode.

### CODE REVIEW

**Review by tellarin — CHANGES_REQUESTED** [review]
Currently fails for "3 minutes from today", which ideally should be a single entity in either mode.
But it's ok to be two separate entities for now.

**Comment by LionbridgeCSII** [comment]
> Currently fails for "3 minutes from today", which ideally should be a single entity in either mode.
> But it's ok to be two separate entities for now.

Modified test cases in Merged according to the corresponding case in Model. 
Strangely in Visual Studio the original test cases (with extracted text 'from today') pass (in Merged) and the updated cases (with extracted text 'today') fail.

File: .NET/Microsoft.Recognizers.Text.DateTime/English/Extractors/EnglishMergedExtractorConfiguration.cs [hunk 1]
```diff
@@ -16,8 +16,9 @@ public class EnglishMergedExtractorConfiguration : BaseDateTimeOptionsConfigurat
         public static readonly Regex AfterRegex =
             new Regex(DateTimeDefinitions.AfterRegex, RegexFlags);
 
-        public static readonly Regex SinceRegex =
-            new Regex(DateTimeDefinitions.SinceRegex, RegexFlags);
+        // used in Experimental mode
+        public static readonly Regex SinceRegexExp =
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/English/Extractors/EnglishMergedExtractorConfiguration.cs:20]: Can you place the two declarations close to each other and have the decision to which one to assign explicitly show the choices? This will help readability.

File: .NET/Microsoft.Recognizers.Definitions.Common/English/DateTimeDefinitions.cs [hunk 2]
```diff
@@ -255,7 +256,7 @@ public static class DateTimeDefinitions
       public static readonly string RelativeDurationUnitRegex = $@"(?:(?:(?<=({NextPrefixRegex}|{PreviousPrefixRegex}|{ThisPrefixRegex})\s+)({DurationUnitRegex}))|((the|my))\s+({RestrictedTimeUnitRegex}))";
       public static readonly string ReferenceDatePeriodRegex = $@"\b{ReferencePrefixRegex}\s+(?<duration>week|month|year|decade|weekend)\b";
       public const string ConnectorRegex = @"^(-|,|for|t|around|@)$";
-      public const string FromToRegex = @"\b(from).+(to)\b.+";
+      public const string FromToRegex = @"(\b(from).+(to)\b.+|\b(from)\s+\d{4}\s+and\s+\d{4})";
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Definitions.Common/English/DateTimeDefinitions.cs:259]: I don't get this regex. Why is this needed?
It seems this is more of a workaround to the issue? Then it seems to be very brittle and will fail for cases like:
"from january and february" or for "from 2002 or 2003".

File: .NET/Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishMergedParserConfiguration.cs [hunk 3]
```diff
@@ -11,7 +11,8 @@ public EnglishMergedParserConfiguration(IDateTimeOptionsConfiguration config)
         {
             BeforeRegex = EnglishMergedExtractorConfiguration.BeforeRegex;
             AfterRegex = EnglishMergedExtractorConfiguration.AfterRegex;
-            SinceRegex = EnglishMergedExtractorConfiguration.SinceRegex;
+            SinceRegex = (config.Options & DateTimeOptions.ExperimentalMode) != 0 ? EnglishMergedExtractorConfiguration.SinceRegexExp :
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishMergedParserConfiguration.cs:14]: Do we need this logic here if it's been properly assigned in EnglishMergedExtractorConfiguration.cs?

File: .NET/Microsoft.Recognizers.Definitions.Common/English/DateTimeDefinitions.cs (hunk 2 above)
Comment (tellarin) [reply .NET/Microsoft.Recognizers.Definitions.Common/English/DateTimeDefinitions.cs:259]: If this regex is just kind of a filter to avoid other cases falling into the ExperimentalMode behaviour, then we should rename it.

**Review by tellarin — CHANGES_REQUESTED** [review]

File: .NET/Microsoft.Recognizers.Text.DateTime/English/Extractors/EnglishMergedExtractorConfiguration.cs (hunk 1 above)
Comment (LionbridgeCSII) [reply .NET/Microsoft.Recognizers.Text.DateTime/English/Extractors/EnglishMergedExtractorConfiguration.cs]: moved declarations close to each other.

File: .NET/Microsoft.Recognizers.Definitions.Common/English/DateTimeDefinitions.cs (hunk 2 above)
Comment (LionbridgeCSII) [reply .NET/Microsoft.Recognizers.Definitions.Common/English/DateTimeDefinitions.cs]: it was meant to exclude only the cases like "from 2002 and 2003". Now I modified it to 
(\b(from).+(to|and|or)\b.+)
but maybe this is too unrestricted?

File: .NET/Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishMergedParserConfiguration.cs (hunk 3 above)
Comment (LionbridgeCSII) [reply .NET/Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishMergedParserConfiguration.cs:14]: yes, as commented in #2230 this seems to be necessary. Otherwise, the regex in MergedParser always gets the standard expression even in Experimental mode.

Comment (LionbridgeCSII) [reply .NET/Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishMergedParserConfiguration.cs:14]: I do not know if there is a different way to do the assignment in the Extractor configuration file which allows to avoid this logic.

Comment (tellarin) [reply .NET/Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishMergedParserConfiguration.cs:14]: Let's see. For now let's use this version.

**Review by tellarin — APPROVED** [review]
